SofaKernel/modules/SofaBaseLinearSolver/CGLinearSolver.inl

```diff
@@ -231,6 +251,20 @@ void CGLinearSolver<TMatrix,TVector>::solve(Matrix& M, Vector& x, Vector& b)
             break;
         }
 
+         if( fabs(den) == 0.0 )
+         {
+             endcond = "div by zero. You scene is probably buggy.";
+             if( verbose )
```

Comment:
Well...

yes but no but yes but no :) 

the only totally redundent thing is:
```cpp
if(printLog.getValue())
     msg_info() << "Blahblah"
```

printLog is the way to activate or deactive *all* the messages going through msg_info() but as it is implemented there is no fine grain control.  This is why adding an additional option like "verbose" (boolean or int) allows to implement multiple level of 'info' verbosity for your component. 

In sofa there is a lot of component that hard code multi-levels verbosity but in my component I prefer to stick to printLog as it make things easier and more consistant.

Reply:
This PR was initially on fixing the failing test...thus I'm not sure it is good if we add too much cleaning or refactoring or cosmetic changes as it will make the review harder.